### PR TITLE
[FLINK-15425][docs] Fix csv format usage in Filesystem connector section

### PR DESCRIPTION
**Pull request by leonardBang**
## What is the purpose of the change

*This pull request aims at fixing document of csv format usage in Filesystem connector section.*

## Brief change log

  - *update file docs/dev/table/connect.md*
  - *update file docs/dev/table/connect.zh.md*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/table/connect.md [hunk 1]
```diff
@@ -663,7 +663,7 @@ CREATE TABLE MyUserTable (
   'connector.type' = 'filesystem',                -- required: specify to connector type
   'connector.path' = 'file:///path/to/whatever',  -- required: path to a file or directory
   'format.type' = '...',                          -- required: file system connector requires to specify a format,
-  ...                                             -- currently only 'csv' format is supported.
+  ...                                             -- currently only 'Old Csv format' is supported if define filesystem connector by DDL.
```
Comment (wuchong) [inline docs/dev/table/connect.md]: The `csv` means `'format.type'='csv'`, so we shouldn't modify it.

File: docs/dev/table/connect.md [hunk 2]
```diff
@@ -676,7 +676,7 @@ CREATE TABLE MyUserTable (
     .path("file:///path/to/whatever")    // required: path to a file or directory
 )
 .withFormat(                             // required: file system connector requires to specify a format,
-  ...                                    // currently only Csv format is supported.
+  ...                                    // currently only 'Csv format' and 'Old Csv format' are supported.
```
Comment (wuchong) [inline docs/dev/table/connect.md]:
```suggestion
  ...                                    // currently only OldCsv format is supported.
```

File: docs/dev/table/connect.md [hunk 3]
```diff
@@ -688,7 +688,7 @@ CREATE TABLE MyUserTable (
     .path("file:///path/to/whatever")  # required: path to a file or directory
 )
 .withFormat(                           # required: file system connector requires to specify a format,
-  ...                                  # currently only Csv format is supported.
+  ...                                  # currently only 'Csv format' and 'Old Csv format' are supported.
```
Comment (wuchong) [inline docs/dev/table/connect.md]:
```suggestion
  ...                                  # currently only OldCsv format is supported.
```

File: docs/dev/table/connect.md [hunk 4]
```diff
@@ -699,7 +699,7 @@ connector:
   type: filesystem
   path: "file:///path/to/whatever"    # required: path to a file or directory
 format:                               # required: file system connector requires to specify a format,
-  ...                                 # currently only "csv" format is supported.
+  ...                                 # currently only 'Old Csv format' is supported if define filesystem connector by YAML.
```
Comment (wuchong) [inline docs/dev/table/connect.md]:
```suggestion
  ...                                 # currently only "csv" format is supported.
```

**Comment by leonardBang** [comment]
@wuchong thanks for you reivew, I update the PR.

File: docs/dev/table/connect.md [hunk 5]
```diff
@@ -676,8 +676,8 @@ CREATE TABLE MyUserTable (
     .path("file:///path/to/whatever")    // required: path to a file or directory
 )
 .withFormat(                             // required: file system connector requires to specify a format,
-  ...                                    // currently only Csv format is supported.
-)                                        // Please refer to Table Formats section for more details.
+  ...                                    // currently only 'csv' format is supported.
```
Comment (wuchong) [inline docs/dev/table/connect.md]: OldCsv ?  Because this is a programming interface.

File: docs/dev/table/connect.md [hunk 6]
```diff
@@ -688,8 +688,8 @@ CREATE TABLE MyUserTable (
     .path("file:///path/to/whatever")  # required: path to a file or directory
 )
 .withFormat(                           # required: file system connector requires to specify a format,
-  ...                                  # currently only Csv format is supported.
-)                                      # Please refer to Table Formats section for more details.
+  ...                                  # currently only 'csv' format is supported.
```
Comment (wuchong) [inline docs/dev/table/connect.md]: OldCsv ?  Because this is a programming interface.

**Pull request merged**